homeassistant/components/here_travel_time/config_flow.py

```diff
@@ -130,36 +139,120 @@ async def async_step_user(
     ) -> FlowResult:
         """Handle the initial step."""
         errors = {}
-        options: dict[str, Any] = {}
         user_input = user_input or {}
         if user_input:
             if self.source == config_entries.SOURCE_IMPORT:
-                user_input, options = self._transform_import_input(user_input)
-                # We need to prevent duplicate imports
-                if any(
-                    is_dupe_import(entry, user_input, options)
-                    for entry in self.hass.config_entries.async_entries(DOMAIN)
-                    if entry.source == config_entries.SOURCE_IMPORT
-                ):
-                    return self.async_abort(reason="already_configured")
+                return await self.async_import(user_input)
             try:
-                await self.hass.async_add_executor_job(validate_input, user_input)
+                await self.hass.async_add_executor_job(
+                    validate_api_key, user_input[CONF_API_KEY]
+                )
+                self._config = user_input
+                return self.async_show_menu(
+                    step_id="user",
+                    menu_options=["origin_coordinates", "origin_entity"],
+                )
             except InvalidCredentialsError:
                 errors["base"] = "invalid_auth"
-            except Exception:  # pylint: disable=broad-except
-                _LOGGER.exception("Unexpected exception")
+            except Exception as error:  # pylint: disable=broad-except
```

Comment:
Catch a more specific exception to intentionally cover each different interesting case

Reply:
At this point there is no anticipated exception beyond the `InvalidCredentialsError`. The API might not be available or a timeout etc.
I could not catch any other error besides the `InvalidCredentialsError` if this will be caught gracefully outside this integration.

Reply:
I removed the broad except and only catch errors thrown by the library. I will open PRs against it to open more transient errors.